### PR TITLE
Add MacOS to Travis CI build

### PR DESCRIPTION
**Pull request by jvaltin**
This adds MacOS to the Travis CI build process. It preps the Travis machine with a Makefile target and then builds the QT target for Mac OS.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -41,7 +41,15 @@ osx-init:
 	export GOPATH=~/go/
 	export PATH=$PATH:~/go/bin
 	make init
-
+osx-init-travis:
```
Comment (benma) [inline Makefile]: No need to have this in the Makefile - can just inline in the travis-ci.sh script?

Comment (jvaltin) [reply Makefile]: OK. Revising and pushing.

File: scripts/travis-ci.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    docker run -i bitbox-wallet-ci bash \
+        -c "make -C \$GOPATH/src/github.com/digitalbitbox/bitbox-wallet-app ci" &
+    while [ -e /proc/$! ]; do echo -n "."  && sleep 60; done
+    docker run --privileged -i bitbox-wallet-ci bash \
+        -c "make -C \$GOPATH/src/github.com/digitalbitbox/bitbox-wallet-app qt-linux"
+    set +e
+fi
+
+if [ "$TRAVIS_OS_NAME" == "osx" ]; then
+    brew install go
```
Comment (benma) [inline scripts/travis-ci.sh]: replace all of this with `make osx-init`?

Comment (jvaltin) [reply scripts/travis-ci.sh]: I think the first step would break as brew is already installed on the system.

File: scripts/travis-ci.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    brew install yarn
+    brew install qt
+    export PATH="/usr/local/opt/qt/bin:$PATH"
+    export LDFLAGS="-L/usr/local/opt/qt/lib"
+    export CPPFLAGS="-I/usr/local/opt/qt/include"
+    export GOPATH=~/go/
+    export PATH=$PATH:~/go/bin
+    mkdir -p /Users/travis/go/src/github.com/digitalbitbox/
```
Comment (benma) [inline scripts/travis-ci.sh]: `mkdir -p $GOPATH/src/github.com/digitalbitbox/`?

File: scripts/travis-ci.sh (hunk 2 above)
Comment (benma) [reply scripts/travis-ci.sh]: Could make it not fail somehow, or just remove it (make a small note in the README that brew is required).

**Comment by jvaltin** [comment]
This is ready for merge and should enable Travis CI for MacOS building. The first commit fails and the second fixes the check script to not fail the build.

**Pull request merged**